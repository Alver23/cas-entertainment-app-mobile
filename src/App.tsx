// Dependencies
import React, { ReactElement } from 'react';
import { ThemeProvider } from 'react-native-elements';
import { SafeAreaView, StatusBar } from 'react-native';

// @Core
import { theme } from './core/theme';

// Container
import { AppIntro } from './containers';

const App = (): ReactElement => {
  return (
    <>
      <StatusBar barStyle="light-content" backgroundColor={theme.colors.blue1} />
      <ThemeProvider theme={theme}>
        <SafeAreaView style={{ flex: 1, backgroundColor: theme.colors.blue1 }}>
          <AppIntro />
        </SafeAreaView>
      </ThemeProvider>
    </>
  );
};

export default App;
