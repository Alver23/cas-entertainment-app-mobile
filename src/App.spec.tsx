// Dependencies
import React from 'react';
import { shallow, ShallowRendererProps } from 'enzyme';

// Containers
import App from './App';

// Mocks
jest.mock('./containers/app', () => ({ AppContainer: jest.fn() }));
jest.mock('./containers/auth-provider', () => ({ AuthProvider: jest.fn() }));

describe('App Component', () => {
  let component: ShallowRendererProps;

  beforeEach(() => {
    component = shallow(<App />);
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
