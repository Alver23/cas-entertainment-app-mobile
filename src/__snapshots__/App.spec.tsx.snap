// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "Button": Object {
          "buttonStyle": Object {
            "height": 50,
            "width": 312,
          },
          "containerStyle": Object {
            "alignItems": "center",
          },
          "titleStyle": Object {
            "color": "#0B132A",
            "fontFamily": "Helvetica",
            "fontSize": 17,
          },
        },
        "Input": Object {
          "containerStyle": Object {
            "alignItems": "center",
          },
          "inputContainerStyle": Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderColor": "#186AA5",
              "borderRadius": 5,
              "borderWidth": 2,
              "height": 48,
              "paddingHorizontal": 10,
              "width": 312,
            },
          ],
          "inputStyle": Object {
            "color": "#495057",
            "fontSize": 17,
          },
        },
        "Text": Object {
          "h4Style": Object {
            "fontSize": 30,
          },
          "style": Object {
            "color": "#ffffff",
            "fontFamily": "Helvetica",
            "fontSize": 15,
          },
        },
        "colors": Object {
          "blue1": "#0B132A",
          "blue2": "#186AA5",
          "color1": "#495057",
          "primary": "#15E1CA",
          "red": "red",
          "transparent": "#F5FCFF88",
          "white": "#ffffff",
        },
      }
    }
  >
    <mockConstructor>
      <ForwardRef(SafeAreaView)
        style={
          Object {
            "backgroundColor": "#0B132A",
            "flex": 1,
          }
        }
      >
        <StatusBar
          animated={false}
          backgroundColor="#0B132A"
          barStyle="light-content"
          showHideTransition="fade"
        />
        <mockConstructor />
      </ForwardRef(SafeAreaView)>
    </mockConstructor>
  </ThemeProvider>
</Provider>
`;
